ykrt/src/compile/jitc_yk/jit_ir.rs

```diff
@@ -487,25 +487,39 @@ impl fmt::Display for LoadInstruction {
 ///
 /// ## Semantics
 ///
-/// Loads a live variable from the trace input struct.
+/// Loads a live variable from the trace input struct. The variable is loaded from the specified
+/// offset (`off`) and the reslting local variable is of the type indicated by the `ty_idx`.
 ///
-/// ## Operands
-///
-/// FIXME: unimplemented as yet.
 #[derive(Debug)]
+#[repr(packed)]
 pub struct LoadArgInstruction {
-    // FIXME: todo
+    /// The byte offset to load from in the trace input struct.
+    off: u32,
+    /// The type of the resulting local variable.
+    ty_idx: TypeIdx,
 }
 
 impl fmt::Display for LoadArgInstruction {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "LoadArg")
+        // FIXME: printing the type (rather than its index) would be better, but that'd require an
+        // overhaul of `Display` to make it more like `aot_or::IRDisplay` (passing the module in,
+        // so we can lookup types).
+        write!(f, "LoadArg {}, {}", self.off(), self.ty_idx.to_usize())
     }
 }
 
 impl LoadArgInstruction {
-    pub(crate) fn new() -> LoadArgInstruction {
-        Self {}
+    pub(crate) fn new(off: u32, ty_idx: TypeIdx) -> LoadArgInstruction {
+        Self { off, ty_idx }
+    }
+
+    pub(crate) fn ty_idx(&self) -> TypeIdx {
+        self.ty_idx
+    }
+
+    pub(crate) fn off(&self) -> u32 {
+        let unaligned = ptr::addr_of!(self.off);
+        unsafe { ptr::read_unaligned(unaligned) }
```

Comment:
It figures it out via type inference of the return type of the function.

What would you want to say in safety? This is always safe AFAIK, because the pointer is in bounds.

Reply:
Because we had to make `LoadArgInstruction` a packed struct to fit all the operands in, the fields became unaligned, and rust doesn't allow reads of unaligned fields.

Reply:
More info: https://doc.rust-lang.org/reference/type-layout.html#the-alignment-modifiers